src/java/voldemort/utils/Props.java

```diff
@@ -128,7 +128,11 @@ public String put(String key, String value) {
     }
 
     public String put(String key, Integer value) {
-        return props.put(key, value.toString());
+        if (null == value) {
```

Comment:
This is a change in the common path, have you examined all the callers to see if someone is not dependent on this behavior.

Also it seems hackish, when setting a value to null, removes the pre-existent value on a base method. It should go in a different method, like reset or remove.